cipher/src/block.rs

```diff
@@ -79,6 +80,20 @@ pub trait BlockCipher {
         }
     }
 
+    /// Encrypt a slice of blocks, leveraging parallelism when available.
+    #[inline]
+    fn encrypt_slice(&self, blocks: &mut [Block<Self>]) {
+        for chunk in blocks.chunks_mut(Self::ParBlocks::to_usize()) {
```

Comment:
I think the following code should be a bit more efficient:
```rust
let pb = Self::ParBlocks::to_usize();
if pb > 1 {
    let mut iter = blocks.chunks_exact_mut(pb);
    for chunk in &mut iter {
        self.encrypt_blocks(chunk.try_into().unwrap())
    }
    blocks = iter.into_remainder();
}
for block in blocks {
    self.encrypt_block(block);
}
```

Reply:
@newpavlov updated the PR to use your suggested implementation.

Agree that ideally the slice-based methods would be called `encrypt_blocks`/`decrypt_blocks`. We can cross that bridge when we get there, I guess.